pyhap/accessories/GarageDoor.py

```diff
@@ -0,0 +1,184 @@
+import time
+import threading
+import pyhap.loader as loader
+import RPi.GPIO as GPIO
```

Comment:
If you use `gpiozero`, then it's possible to "swap out" the pin factory - this makes testing from another machine (where you have `pigpio` remote pins).

Reply:
Will look into it tomorrow or next. Need to rest  :)

Reply:
https://bitbucket.org/schinckel/garagedoor/src/e3268a75ca576f7dc65532adf1f6d51a8bc149fc/garage_door/accessory.py?at=default&fileviewer=file-view-default

Reply:
Oh, the other great thing about `gpiozero` is that you can set event handlers on pins.

In other words: "when the top trigger opens, this means the door is now closing", and so on.

Reply:
Yeah, thats basically what the callbacks are doing using the GPIO library. I only have one switch on my door, so I infer a lot here, on state and transition. lol.

Reply:
My excuse to learn a lot of different frameworks.

https://gitlab.com/jslay/garage_hub

Reply:
I'm happy to have one that works (a) with no switches: time delay based only, (b) a single switch, and (c) switches at both the top and bottom.

Reply:
yeah, thats one thing i was thinking about, how best to make this Accessory dynamic. 

I just started getting confused about what was happening with my issues listed above, that I figured I'd go ahead and post what I had so far.